common/core/nesteddisplay.cpp

```diff
@@ -207,4 +372,48 @@ bool NestedDisplay::GetDisplayName(uint32_t *size, char *name) {
   return true;
 }
 
+int NestedDisplay::start_sock_service() {
+  struct sockaddr_in server_addr;
+  msock_fd = socket(AF_INET, SOCK_STREAM, 0);
+  if (msock_fd < 0) {
+    ETRACE("Cannot create socket fd:%d",msock_fd);
+    return -1;
+  }
+
+  memset(&server_addr, 0, sizeof(server_addr));
+  int portno = 2345;
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_addr.s_addr = INADDR_ANY;
+  server_addr.sin_port = htons(portno);
+
+  if (bind(msock_fd, (struct sockaddr *) &server_addr, sizeof(server_addr)) < 0) {
+    close(msock_fd);
+    msock_fd = -1;
+    return -1;
+  }
+
+  listen(msock_fd, 1);
+  signal(SIGPIPE, signal_callback_handler);
+  return 0;
+}
+
+int NestedDisplay::hyper_communication_network_send_data(void *data, int len)
+{
+  if (mclient_sock_fd >= 0) {
+    return send(mclient_sock_fd, data, len, 0);
+  }
+  return -1;
+}
+
+
```

Comment:
un-necessary extra line

Reply:
they are necessary code, if socket client(dom0 side) lost connect with  socket server(hwc side), then hwc continue send metadata to client will trigger this handler, then hwc will wait for another new connection from dom0 side

Reply:
extra new lines I mean which are empty

Reply:
ok, will delete it